internal/db/db_certificate_authorities.go

```diff
@@ -515,9 +517,16 @@ func (db *Database) SignCertificateRequest(csrFilter CSRFilter, caFilter Certifi
 		return err
 	}
 	if CSRIsForACertificateAuthority {
-		err := db.UpdateCertificateAuthorityCertificate(ByCertificateAuthorityID(caToBeSigned.CertificateAuthorityID), certPEM.String()+caRow.CertificateChain)
-		if err != nil {
-			return err
+		if wasSelfSigned {
+			err := db.UpdateCertificateAuthorityCertificate(ByCertificateAuthorityID(caToBeSigned.CertificateAuthorityID), certPEM.String()+certPEM.String())
+			if err != nil {
+				return err
+			}
+		} else {
+			err := db.UpdateCertificateAuthorityCertificate(ByCertificateAuthorityID(caToBeSigned.CertificateAuthorityID), certPEM.String()+caRow.CertificateChain)
+			if err != nil {
+				return err
+			}
```

Comment:
Are we allowing a CA that was self signed to be signed now? If so could you please add this part and a rationale to the description of the PR?